internal/jobtracker/jobtracker_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 Canonical.
+package jobtracker_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/jobtracker"
+	"github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type jobTrackerSuite struct {
+	db      *db.Database
+	tracker *jobtracker.Tracker
+}
+
+func (s *jobTrackerSuite) Init(c *qt.C) {
+	db := &db.Database{
+		DB: jimmtest.PostgresDB(c, time.Now),
+	}
+	err := db.Migrate(context.Background())
+	c.Assert(err, qt.IsNil)
+
+	s.db = db
+	tracker, err := jobtracker.NewJobTracker(db, time.Second*5)
+	c.Assert(err, qt.IsNil)
+	s.tracker = tracker
+}
+
+// !!! --- For PR reviewers, currently we're polling the job status as we are waiting
```

Comment:
for PR author - remove this comment and simply comment on your own PR to get the message across to reviewers ;)

Reply:
Yep fair point 😆